consensus/utils.go

```diff
@@ -6,9 +6,51 @@ import (
 
 	"github.com/deso-protocol/core/bls"
 	"github.com/deso-protocol/core/collections"
+	"github.com/holiman/uint256"
 	"golang.org/x/crypto/sha3"
 )
 
+// Given a QC and a sorted validator set, this function returns true if the QC contains a valid
+// super-majority of signatures from the validator set for the QC's (View, BlockHash) pair.
+func IsValidSuperMajorityQuorumCertificate(qc QuorumCertificate, validators []Validator) bool {
+	if !isProperlyFormedQC(qc) || !isProperlyFormedValidatorSet(validators) {
+		return false
+	}
+
+	// Compute the signature that validators in the QC would have signed
+	signaturePayload := GetVoteSignaturePayload(qc.GetView(), qc.GetBlockHash())
+
+	// Compute the total stake in the QC and the total stake in the network
+	stakeInQC := uint256.NewInt()
+	totalStake := uint256.NewInt()
+
+	// Fetch the validators in the QC
+	validatorPublicKeysInQC := []*bls.PublicKey{}
+
+	// Fetch the validators in the QC, and compute the sum of stake in the QC and in the network
+	for ii := range validators {
+		if qc.GetSignersList().Get(ii) {
+			stakeInQC.Add(stakeInQC, validators[ii].GetStakeAmount())
+			validatorPublicKeysInQC = append(validatorPublicKeysInQC, validators[ii].GetPublicKey())
+		}
+		totalStake.Add(totalStake, validators[ii].GetStakeAmount())
+	}
+
+	// Check if the QC contains a super-majority of stake
+	if !isSuperMajorityStake(stakeInQC, totalStake) {
+		return false
+	}
+
+	// Finally, validate the signature
+	isValidSignature, err := bls.VerifyAggregateSignatureSinglePayload(
```

Comment:
I'm getting a warning here that `isValidSignature` is already defined. Should we maybe rename it so it doesn't overlap with an existing method?

Reply:
Oof good catch. There was a function with the same name. Fixed now